src/Features/Core/Portable/Completion/Providers/ImportCompletionProvider/AbstractTypeImportCompletionService.CacheEntry.cs

```diff
@@ -41,15 +45,21 @@ internal abstract partial class AbstractTypeImportCompletionService
                 string genericTypeSuffix,
                 bool isInternalsVisible,
                 bool isAttributeContext,
-                bool isCaseSensitive)
+                bool isCaseSensitive,
+                bool hideAdvancedMembers)
             {
                 // We will need to adjust some items if the request is made in:
                 // 1. attribute context, then we will not show or complete with "Attribute" suffix.
                 // 2. a project with different language than when the cache entry was created,
                 //    then we will change the generic suffix accordingly.
+                // 3. option to show advanced members is false and there is advanced member items in this cache entry
+                //    (which need to be filtered out)
                 // Otherwise, we can simply return cached items.
                 var isSameLanguage = Language == language;
-                if (isSameLanguage && !isAttributeContext)
+                var needToFilterOutAdvancedMembers = hideAdvancedMembers && ContainsAdvancedMembers;
+                if (isSameLanguage &&
+                    !isAttributeContext &&
+                    !needToFilterOutAdvancedMembers)
                 {
                     return ItemInfos.Where(info => info.IsPublic || isInternalsVisible).SelectAsArray(info => info.Item);
                 }
```

Comment:
i realized what i found odd about thsi.  you have effectively two loops over ItemInfos to decide which to return.  instead of that, why not have just one loop and do all the checks within it...  i don't really see the benefit of doing it this way.

Reply:
also, updat ethe code a few lines below to be `using var _ = ArrayBuilder...`

Reply:
This way we do fewer checks inside the first loop, which accounts for majority cases, but in terms of perf benefit, probably not much (but hey, it makes me feel better somehow :D)

Reply:
i don't really see that being hte case.  if you perform the bottom loop, all that will happen is the check for `hideAdvancedMembers && info.IsEditorBrowsableStateAdvanced` (two bools), and the check for `if (isAttributeContext)` (one bool) and the check of `if (!isSameLanguage && info.IsGeneric)`  two more bools.  This should be nothing :)

Reply:
note:  the code should doc this condition:

```
if (!isSameLanguage && info.IsGeneric)
```

why are we not caching generics across languages?

Reply:
Because in C# and VB the display text is different for generics, i.e. `<T>` and `(Of T)`, and for simpllicity, we only cache for one language

Reply:
(that's why we should have a comment in hte code) :)